roles/aap_rules_validation/tasks/check_roles.yml

```diff
@@ -0,0 +1,132 @@
+---
+- name: Init variables
+  ansible.builtin.set_fact:
+    __forbidden_roles: []
+    __forbidden: []
+    __objects_and_roles: []
+    __objects_and_roles_product: []
+    unallowed_roles: []
+
+- name: Verify forbidden roles presence
+  when: rule['forbidden_roles'] is defined
+        and rule['forbidden_roles'] != None
+        and rule['forbidden_roles'] | length > 0
+  block:
+
+    - name: Set forbidden roles list
+      ansible.builtin.set_fact:
+        __forbidden_roles: "{{ __forbidden_roles + ([item.key] | product(item.value)) }}"
+      with_dict: "{{ rule['forbidden_roles'] }}"
```

Comment:
The `with_dict` is considered legacy. Should use `loop: "{{ rule['forbidden_roles'] | dict2items }}"` instead.

See https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_loops.html#with-dict

Reply:
thanks !
done in [980ea8a3](https://github.com/redhat-cop/aap_configuration_extended/pull/16/commits/980ea8a3d4b56e441ef635ab4d623bce0b59ad3d)